tools/scripts/checkLicense.py

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env python3
+import os
+import sys
+
+
+def get_license():
+    f = open("src/catch2/catch_all.hpp", "r")
+    license = "".join(f.readlines()[0:7])
+    f.close()
+    return license
+
+
+def check_license(license):
+    failed = 0
+    base_dir = "src/catch2/"
+
+    # The _ represents the list of directories in base_dir
+    for root, _, files in os.walk(base_dir):
+        for file in files:
+            f = open(root + "/" + file, "r")
+            file_license = "".join(f.readlines()[0:7])
```

Comment:
Thinking about this some more:
Why turn this into a string? Python lists are comparable, so you can just compare the slices directly.

Reply:
Well managing one big string is easier than managing many small ones for me. But you are right the code is probably better without the `join`.